### ISSUE TITLE
I'm wishing for: Use ds18b20 sensor and add local 128x64 IC2 OLED display

### ISSUE THREAD

**Issue by VastSky** [Schm1tz1/ESPressIoT#11]
Roman,

I have been enjoying EspressIoT but am miserable because I'll never be able to learn to code well enough to add a OLED display to this project. I have spent countless hours searching to try to understand how to add U8g2 to support the OLED and failed. I like the tab layout of the sketch, but I believe this may be adding an insurmountable challenge to my lack of ability.

Any plans to return to this great project of yours? Hoping you might release a ds18b20, 
128x64 IC2 OLED version!

Best regards,

Ed Rilkoff - Kamloops BC Canada

**Comment by Schm1tz1** [comment]
Hi, the ds18b20 should work pretty good, you only might need to adjust the readout times.
I did my first controller "cofcon" with dallas sensors I started working with but I changes to TSIC because of their higher accuracy and faster readout. 
Maybe have a look at this code - should be pretty easy to impmement:
https://github.com/Schm1tz1/cofcon/blob/master/cofcon/sensors_dallas.ino

Concerning the OLED display: What display are you using? Do you have a link? Did you have a look at the u8g2-library (https://github.com/olikraus/u8g2)?
I did not work with displays so far but that might be interesting.

**Comment by VastSky** [comment]
Roman,

Thanks for responding!! - Mind Blown!!   

I got EspressIoT implemented in my machine about two years ago with DS18B20. I was able to merge the DS18B20 sensor from Cofcon into EspressIoT largely because you have such consistent code structure. I simply copied the DS18B20 tab from cofcon into EspressIoT and substituted any TSIC references, and deleted the original TSIC tab to get it to compile. The DS18B20 values return at a high rate when viewing the MQTT stream. Your cofcon DS18B20 variables seem to work well. I think this sensor is quick enough but may not survive steaming temperatures. 

Your web interface works great but is a bit slow to update on my iPhone. In the morning and I'm getting ready for work I simply would love to glance at a small display right on my Rancilio Silvia to confirm that the boiler temperature is ready to brew. A Generic 0.96" SSD1306 I2C 128x64 OLED display is perfect as it is a similar size as a Wemos D1 mini. The display and a Wemos D1 fit nice together in a 3D printed case.

Display: Ebay: Generic SSD1306 0.96", 128x64, I2C OLED 
uses 3 to 5vdc, 
GND, 
SDAto D2,  
SCL to D1

Typical: https://www.ebay.ca/sch/i.html?_from=R40&_trksid=m570.l1313&_nkw=++0.96%09+0.96%22+I2C+IIC&_sacat=0

http://wiki.sunfounder.cc/index.php?title=OLED-SSD1306_Module

I did get the G8g2 examples to work with an OLED display and Wemos D1, but merging this in to the EspressIoT tab structure is what I could not figure out how to implement. 

MQTT coming from EspressIoT is seen by "MQTT Explorer" - a great free utility https://github.com/thomasnordquist/MQTT-Explorer Perhaps I will search how to plot inputTemp values on a graph. 

Adding code to support MQTT authentication credentials in order to connect a MQTT server would also be great. I could send values to free MQTT cloud servers! 

If you like I would be happy to order a few of these displays for you to try, and have them shipped to you. Let me know.

Roman - Thank YOU for a fantastic project!!

**Comment by Schm1tz1** [comment]
Hi Ed,

very nice! Good to hear that you implemented it. I'm pretty busy at the moment and need maybe one Saturday afternoon to make a test setup and play with this. I think I had it implemented DS18B20 in an old version but not sure if I ever pushed it to Github to be honest. I will create an "experimental" branch, so feel free to contribute your code. I can also start with some "skeleton" for the display so you know how to use it. 
No worries ordering a display - it's pretty cheap and I wanted one to play with anyway ;-). I ordered one at ebay and it arrived yesterday evening ;-).
For MQTT we have to check - I have used the PubSub-Library as it is pretty lightweight and easy to use. I think it can handle user/pass authentication, so one could configure that and store in the JSON config that is saved to SPIFFS.

Cheers,
Roman
